src/include/expression/constant_value_expression.h

```diff
@@ -59,12 +97,7 @@ class ConstantValueExpression : public AbstractExpression {
 
   virtual void Accept(SqlNodeVisitor *v) override { v->Visit(this); }
 
-  virtual hash_t Hash() const override {
-    hash_t hash = HashUtil::Hash(&exp_type_);
-    return HashUtil::CombineHashes(hash, value_.Hash());
-  }
-
-  bool IsNullable() const override { return false; }
+  bool IsNullable() const override { return value_.IsNull(); }
```

Comment:
Is this correct? If you're extracting constants out, then NULLness can change between invocations?

Reply:
This is correct. This only fixes the bug in `IsNullable()`, and not related to parameterization.

The constant values are always considered as non-nullable.  Please see `VisitParameters()` in `constant_value_expression.h` below, which creates parameter information for this constant value expression and specifies its nullability implicitly at `CreateConstParameter()` - default is `true` here. I might specify this explicitly for code readability.

```
  virtual void VisitParameters(std::vector<Parameter> &parameters,
      std::unordered_map<const AbstractExpression *, size_t> &index,
      UNUSED_ATTRIBUTE const std::vector<peloton::type::Value>
          &parameter_values) override {
    // Add a new parameter object for a constant
    parameters.push_back(Parameter::CreateConstParameter(GetValue()));
    index[this] = parameters.size() - 1;
  };
```